application/src/main/java/org/thingsboard/server/service/security/auth/TokenOutdatingService.java

```diff
@@ -17,65 +17,44 @@
 
 import io.jsonwebtoken.Claims;
 import lombok.RequiredArgsConstructor;
-import org.springframework.cache.Cache;
-import org.springframework.cache.CacheManager;
 import org.springframework.context.event.EventListener;
 import org.springframework.stereotype.Service;
-import org.thingsboard.server.common.data.CacheConstants;
+import org.springframework.util.StringUtils;
+import org.thingsboard.server.cache.TbTransactionalCache;
 import org.thingsboard.server.common.data.id.UserId;
 import org.thingsboard.server.common.data.security.event.UserAuthDataChangedEvent;
 import org.thingsboard.server.common.data.security.model.JwtToken;
-import org.thingsboard.server.config.JwtSettings;
 import org.thingsboard.server.service.security.model.token.JwtTokenFactory;
 
-import javax.annotation.PostConstruct;
 import java.util.Optional;
 
 import static java.util.concurrent.TimeUnit.MILLISECONDS;
-import static java.util.concurrent.TimeUnit.SECONDS;
 
 @Service
 @RequiredArgsConstructor
 public class TokenOutdatingService {
-    private final CacheManager cacheManager;
+    private final TbTransactionalCache<String, Long> cache;
     private final JwtTokenFactory tokenFactory;
-    private final JwtSettings jwtSettings;
-    private Cache usersUpdateTimeCache;
-
-    @PostConstruct
-    protected void initCache() {
-        usersUpdateTimeCache = cacheManager.getCache(CacheConstants.USERS_UPDATE_TIME_CACHE);
-    }
 
     @EventListener(classes = UserAuthDataChangedEvent.class)
     public void onUserAuthDataChanged(UserAuthDataChangedEvent event) {
-        usersUpdateTimeCache.put(toKey(event.getUserId()), event.getTs());
+        if (StringUtils.hasText(event.getId())) {
+            cache.put(event.getId(), event.getTs());
+        }
     }
 
     public boolean isOutdated(JwtToken token, UserId userId) {
         Claims claims = tokenFactory.parseTokenClaims(token).getBody();
         long issueTime = claims.getIssuedAt().getTime();
+        String sessionId = claims.get("sessionId", String.class) == null ? "" : claims.get("sessionId", String.class);
+        return isTokenOutdated(issueTime, userId.toString()) || isTokenOutdated(issueTime, sessionId);
```

Comment:
Also, sessionId might be "" and you will execute one more cache lookup. Why? This is extremely inefficient.